esque/message.py

```diff
@@ -0,0 +1,81 @@
+import json
+import pathlib
+from typing import Iterable, NamedTuple, Any
+
+from confluent_kafka.cimpl import Message
+
+
+class DecodedMessage(NamedTuple):
+    key: str
+    value: str
+    partition: int
+
+
+class KafkaMessage(NamedTuple):
+    key: Any
+    value: Any
+    partition: int
+    key_schema: str = None
+    value_schema: str = None
+
+
+class IOHandler:
+    def __init__(self, directory: pathlib.Path):
+        self.directory = directory
+        self.file_name = "data"
+        self.open_mode = "w+"
+
+    def __enter__(self):
+        if not self.directory.exists():
+            self.directory.mkdir()
+        self.file = (self.directory / self.file_name).open(self.open_mode)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.file.close()
+
+
+class FileWriter(IOHandler):
+    def write_message_to_file(self, message: Message):
+        pass
+
+
+class FileReader(IOHandler):
+    def read_from_file(self) -> Iterable[KafkaMessage]:
+        pass
+
+
+class PlainTextFileWriter(FileWriter):
+    def __init__(self, directory: pathlib.Path):
```

Comment:
You could move this whole constructor to `FileWriter`, I guess all writers need that mode.